### PR TITLE
Add completions for possible imports

### LINKED ISSUE #241: Auto import completions
Following up from #230, there should also be completion suggestions for possible imports. On selected, it should automatically add the import.

### PR DESCRIPTION
**Pull request by jmbockhorst**
An WIP for adding completions for possible imports. 

### Update
This works fairly well now. By doing some custom filtering and limiting the completion items to 50 at a time and marking it as incomplete if there are still more, I was able to increase the performance and make it usable in a large workspace.

### Still to do:
- Add some tests

Closes #241

### CODE REVIEW

**Comment by jmbockhorst** [comment]
@Razzeee This should be ready for you to take a look at the approach and what you think. Thanks.

**Comment by Razzeee** [comment]
Are you aware of the ESLint errors?

**Comment by jmbockhorst** [comment]
I had a problem with the vscode eslint extension causing this error on every source line : `Delete `␍`eslintprettier/prettier`, so I had temporarily disable it. Not sure what's going on there.

**Comment by jmbockhorst** [comment]
I guess I needed to switch my line endings to `LF`.

File: src/test/completionProvider.test.ts [hunk 1]
```diff
@@ -456,4 +479,94 @@ func msg =
...
+
+--@ Test.elm
+module Test exposing (..)
+
+func : String
+func = 
+  {-caret-}
+`;
+
+    // await testCompletions(source2, [
```
Comment (Razzeee) [inline src/test/completionProvider.test.ts:539]: Should this still be commented out?

Comment (jmbockhorst) [reply src/test/completionProvider.test.ts:539]: It’s a failing test, so yes until I fix the source.

**Comment by Razzeee** [comment]
Thank you, as a heads up, I'll try to release a new version this weekend, as we haven't released for quiet some time.

**Comment by jmbockhorst** [comment]
Thanks. I will work on some more clean up in this area tonight and tomorrow.

**Pull request merged**